modules/renter/contractor/editor.go

```diff
@@ -130,9 +140,19 @@ func (he *hostEditor) Upload(data []byte) (crypto.Hash, error) {
 }
 
 // Delete negotiates a revision that removes a sector from a file contract.
-func (he *hostEditor) Delete(root crypto.Hash) error {
+func (he *hostEditor) Delete(root crypto.Hash) (err error) {
 	he.mu.Lock()
 	defer he.mu.Unlock()
+
+	// Increase Successful/Failed interactions accordingly
+	defer func() {
+		if err != nil {
+			he.contractor.hdb.IncrementFailedInteractions(he.contract.HostPublicKey)
+		} else {
+			he.contractor.hdb.IncrementSuccessfulInteractions(he.contract.HostPublicKey)
+		}
+	}()
+
 	if he.invalid {
 		return errInvalidEditor
```

Comment:
This check needs to happen after the deferred func, because it's not a fault of the host - `he.invalid` indicates I believe that the renter has done a renewal or is otherwise no longer using the contract. So we wouldn't want to increment the reliability stats at all.